File: SDKs/netboxdemo.com/2.8/python/src/sdk/models/operations/dcim_device_roles_create.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared


@dataclass
class DcimDeviceRolesCreateRequest:
    request: shared.DeviceRole = field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    

@dataclass
class DcimDeviceRolesCreateResponse:
    content_type: str = field(default=None)
    device_role: Optional[shared.DeviceRole] = field(default=None)
    status_code: int = field(default=None)
    

File: SDKs/netboxdemo.com/2.8/go/pkg/models/operations/dcimdevicerolescreate.go
package operations

import (
"openapi/pkg/models/shared")

type DcimDeviceRolesCreateRequest struct {
    Request shared.DeviceRole `request:"mediaType=application/json"`
    
}

type DcimDeviceRolesCreateResponse struct {
    ContentType string 
    DeviceRole *shared.DeviceRole 
    StatusCode int64 
    
}

